Improve proposals answer page

mrcasals opened the pull request:
#### :tophat: What? Why?
This PR adds a proposal show page for the admin section. this page lists some proposal data, and merges the proposal notes and answer pages into it. This means we can get rid of the notes and answer pages, since they have been merged into the show page.

#### :pushpin: Related Issues
- Related to #5628
- Fixes #5630

#### :clipboard: Subtasks
- [x] Add `CHANGELOG` entry
- [x] Add show page with proposal info
- [x] Move proposal notes to the show page
- [x] Move proposal answer to the show page
- [x] Add rankings
- [x] Show related meetings
- [x] Show attachments
- [x] Find a way to start a conversation with the author
- [x] Fix tests

### :camera: Screenshots (optional)
![image](https://user-images.githubusercontent.com/491891/74138318-af642980-4bf1-11ea-876e-50e3ac4f8924.png)

If the current admin can edit the proposal, a link to edit it is shown next to the proposal title: 
![image](https://user-images.githubusercontent.com/491891/74138296-a7a48500-4bf1-11ea-95a0-bf257cc960d3.png)

mrcasals commented:
@decidim/core tests fail due to something about CodeClimate, can you check it please? I'm not sure what's going on...

leio10 commented on decidim-proposals/app/views/decidim/proposals/admin/proposals/show.html.erb:
```diff
@@ -74,4 +74,6 @@
 </div>
 
-<%= render "decidim/proposals/admin/proposal_notes/proposal_notes" %>
+<% if allowed_to? :create, :proposal_note %>
```
Shouldn't admins who can't create proposal notes be allowed to view existing notes?

mrcasals replied:
I kept the current behavior, and currently if you can't create proposal notes you can't view the current ones...

leio10 commented on Gemfile.lock:
```diff
@@ -684,10 +685,11 @@ GEM
       rubyzip (>= 1.2.2)
     seven_zip_ruby (1.2.5)
-    simplecov (0.18.0)
+    simplecov (0.17.1)
```
Jeje, I thought this would happen when I saw the first commit 😄

mrcasals replied:
Yeah, I had a little problem with dependencies and rebasing from master 😞

leio10 replied:
Oh, I had problems with CodeClimate coverage report when I updated to simplecov 0.18, I thought that was the problem: https://github.com/codeclimate/test-reporter/issues/418

mrcasals replied:
Yes, we had to lock `simplecov` to v0.17.1 so that CI jobs could end successfully, see #5673 😄

tramuntanal commented on decidim-proposals/app/views/decidim/proposals/admin/proposals/show.html.erb:
```diff
@@ -0,0 +1,143 @@
+
+    <div class="row column">
+      <strong><%= t ".related_meetings" %>:</strong>
+        <ul id="related-meetings">
```
When linking a meeting to a proposal
![imatge](https://user-images.githubusercontent.com/199462/74109543-9a04e600-4b84-11ea-831c-8f325ae94681.png)

I can't see the meeting listed:
![imatge](https://user-images.githubusercontent.com/199462/74109556-b86ae180-4b84-11ea-93f6-92f87968aca8.png)

Can it be something related to my local environment?

tramuntanal replied:
mmm.. I think this is because of this: https://github.com/decidim/decidim/pull/5674
so, ok

mrcasals replied:
That's correct, it's due to #5674!

mrcasals commented on decidim-proposals/app/views/decidim/proposals/admin/proposals/_proposal-tr.html.erb:
```diff
@@ -56,15 +56,7 @@
-    <% else %>
-      <%= icon "comment-square", scope: "decidim.proposals.admin", class: "action-icon action-icon--disabled" %>
-    <% end %>
+    <%= icon_link_to "comment-square", proposal_path(proposal), t("actions.show", scope: "decidim.proposals"), class: "icon--small" %>
```
Product agreed on using that icon, see discussion on the related issue.

mrcasals replied:
Thinking back on this, do you mind if we merge this PR and in the meantime I approach product to rethink the icon? I'll keep the discussion in the issue 😄

tramuntanal replied:
Ok, let's merge and you decide with @decidim/product which icon to use, i recommend the info icon as mentioned before, but don't have an exhaustive knowledge of the icons used in the whole admin.

mrcasals commented on decidim-proposals/app/helpers/decidim/proposals/admin/proposals_helper.rb:
```diff
@@ -15,6 +15,20 @@ def meetings_as_authors_selected
+            if identity.is_a?(Decidim::Organization)
+              Decidim::Proposals::OfficialAuthorPresenter.new
+            else
+              present(identity)
```
It should, it's using the `present` helper method we use in the public area...

mrcasals commented on decidim-comments/app/models/decidim/comments/comment.rb:
```diff
@@ -39,6 +39,18 @@ class Comment < ApplicationRecord
 
       delegate :organization, to: :commentable
 
+      def self.positive
```
They are equivalent, and we've used this approach throughout the Decidim codebase 😄

mrcasals commented:
@tramuntanal I've added the link to the edit page (see screenshot in the PR description), and I've updated the screenshot relating a meeting with the PR, so that it shows they appear.

Regarding the icon, as said in https://github.com/decidim/decidim/pull/5671#discussion_r376933855, I've moved the discussion to #5630, but please let's not block this PR for that, I'll open a new one if needed 🙏 

Can you re-review this please?
